src/Symfony/Component/Serializer/Encoder/XmlEncoder.php

```diff
@@ -335,7 +336,7 @@ private function parseXmlValue(\DOMNode $node, array $context = array())
         }
 
         foreach ($value as $key => $val) {
-            if (is_array($val) && 1 === count($val)) {
+            if (is_array($val) && 1 === count($val) && (!$this->forceCollection || !is_array($val[0]))) {
```

Comment:
After a second thought, wouldn't it be better to allow to change this behavior through the context instead of forcing to register a new instance of the class?

Reply:
I think it will be easier to use if we don't create a new instance. Ill do the modification on both PRs